Hazel/src/Hazel/Renderer/Renderer2D.cpp

```diff
@@ -56,7 +57,8 @@ namespace Hazel {
 			{ ShaderDataType::Float4, "a_Color" },
 			{ ShaderDataType::Float2, "a_TexCoord" },
 			{ ShaderDataType::Float, "a_TexIndex" },
-			{ ShaderDataType::Float, "a_TilingFactor" }
+			{ ShaderDataType::Float, "a_TilingFactor" },
+			{ ShaderDataType::Float2, "a_UVOffsetFactor" }
```

Comment:
I don't think you need to modify the shader and buffer layout, the texture coordinate could be offset in the renderer directly with `s_Data.QuadVertexBufferPtr->TexCoord = textureCoords[i] - uvOffsetFactor;`

Reply:
Oh yes, that's true, I don't know how I didn't see that